cunumeric/eager.py

```diff
@@ -485,21 +486,49 @@ def swapaxes(self, axis1: int, axis2: int) -> NumPyThunk:
         self.children.append(result)
         return result
 
-    def convert(self, rhs: Any, warn: bool = True) -> None:
+    def convert(
+        self,
+        rhs: Any,
+        warn: bool = True,
+        nan_op: Optional[int] = None,
+    ) -> None:
         self.check_eager_args(rhs)
         if self.deferred is not None:
             return self.deferred.convert(rhs, warn=warn)
         else:
             if self.array.size == 1:
-                self.array.fill(rhs.array.item())
+                if nan_op is ConvertCode.SUM and np.isnan(rhs.array.item()):
+                    self.array.fill(0)
+                elif nan_op is ConvertCode.PROD and np.isnan(rhs.array.item()):
+                    self.array.fill(1)
+                else:
+                    self.array.fill(rhs.array.item())
             else:
                 if (
                     rhs.array.dtype.kind == "c"
                     and self.array.dtype.kind != "c"
                 ):
-                    self.array[:] = rhs.array.real
+                    if nan_op is ConvertCode.SUM:
+                        self.array[:] = np.select(
```

Comment:
you wouldn't have to wrap the arguments with brackets if you used `np.where`.

Reply:
661ee85e10d7543c346b087a05851e3326a05700